pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -35,24 +36,32 @@ type S3Agent struct {
 	Client *s3.S3
 }
 
-func NewS3Agent(accessKey, secretKey, endpoint string, debug bool) (*S3Agent, error) {
+func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, TLSEnabled bool) (*S3Agent, error) {
 	const cephRegion = "us-east-1"
 
 	logLevel := aws.LogOff
 	if debug {
 		logLevel = aws.LogDebug
 	}
+	client := http.Client{
+		Timeout: time.Second * 15,
+	}
+	if TLSEnabled {
+		tr := &http.Transport{
+			// #nosec G402 since S3 client running inside RGW pod
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Why would we need to use this unsafe mode? If we have signed certificates we don't need that.

Reply:
I don't understand your reply. I guess my point is that we should not use `InsecureSkipVerify: true` if we have signed-certificates by known CAs.